src/intern.rs

```diff
@@ -224,6 +237,15 @@ impl<T: Eq + Hash + Send + Sync + 'static> Intern<T> {
             Intern { pointer: p }
         })
     }
+    /// Check if a value already is interned.
+    ///
+    /// If this value has previously been interned, return true, else returns false
+    pub fn is_interned<'a, Q: ?Sized + Eq + Hash + 'a>(val: &'a Q) -> bool
+    where
+        &'static T: Borrow<Q> + From<&'a Q>,
```

Comment:
**suggestion:** You can remove the `From` constraint for this method.

```suggestion
    pub fn is_interned<'a, Q: ?Sized + Eq + Hash + 'a>(val: &'a Q) -> bool
    where
        &'static T: Borrow<Q>,
```

Reply:
If I do, then I need to specify generic bounds on line 72:

```
type annotations needed
cannot satisfy `_: std::cmp::Eq`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B24%5D?24#file%3A%2F%2F%2Fhome%2Fdanieljoyce%2Fgit%2Fgeneral%2Finternment%2Fsrc%2Fintern.rs)
intern.rs(204, 9): required by a bound in `Intern::<T>::is_interned`
intern.rs(72, 19): consider specifying the generic argument: `::<T>`
```

Reply:
Okay, one sec.

Reply:
If I remove the From<> bounds on T, then it requires dismbiguation per the above error message.

Reply:
If I move the is_interned to the `impl<T: Eq + Hash + Send + Sync + 'static + ?Sized> Intern<T> {` block, then

```
 let _ = Intern::new("Fortunato");
    assert!(Intern::is_interned(&"Fortunato"));
    assert!(Intern::is_interned("Fortunato"));
```

Both check as valid, but when I run the test, ` assert!(Intern::is_interned("Fortunato"));` fails.